sdk/core/azure-core/azure/core/tracing/_models.py

```diff
@@ -0,0 +1,69 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from enum import Enum
+from typing import Dict, Mapping, Optional, Union, Sequence, TypedDict
+
+
+AttributeValue = Union[
+    str,
+    bool,
+    int,
+    float,
+    Sequence[str],
+    Sequence[bool],
+    Sequence[int],
+    Sequence[float],
+]
+Attributes = Mapping[str, AttributeValue]
+
+
+class SpanKind(Enum):
```

Comment:
Should we use CaseInsensitiveEnum here?

Reply:
Ah I see, I guess if we were creating this enum today maybe we would make case insensitive. I agree that it's more useful for string-valued enums, but it I think it should allow case insensitivity on the keys as well, e.g. SpanKind.client